docs/DeveloperGuide.md

```diff
@@ -288,31 +288,215 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+   * Use case resumes from step 2. 
+
+* 1c. User wants to edit the name field to a name that already exists in PoochPlanner.
+
+   * 1c1. PoochPlanner displays the error message.
+   * 1c2. User re-enters the command with a diferent name.
+   * Steps 1c1 - 1c2 are repeated until the new name field is valid.
+   * Use case resumes from step 2. 
+
+* 1d. User did not specify the field that they want to edit.
+
+   * 1d1. PoochPlanner displays the error message.
+   * 1d2. User re-enters the command and specify the field/s to edit.
+   * Steps 1d1 - 1d2 are repeated until there exist a specified field to edit.
+   * Use case resumes from step 2. 
+
+
+* 1e. User specified an invalid field.
+
+   * 1e1. PoochPlanner displays the error message.
+   * 1e2. User re-enters the command and edits a different field.
+   * Steps 1e1 - 1e2 are repeated until there exist a valid field in the input.
+   * Use case resumes from step 2. 
+---
+**System**: `PoochPlanner`
 
-  Use case ends.
+**Use case**: `UC04 - Search Contact of a Person`
 
-* 3a. The given index is invalid.
+**Actor**: `User`
 
-    * 3a1. AddressBook shows an error message.
+**MSS**:
 
-      Use case resumes at step 2.
+1.  User chooses to search for the contact of a person with a keyword for a specified field.
+2.  PoochPlanner confirms successful search.
+3.  PoochPlanner returns the sublist of contacts that contains the keyword specified by the user. 
 
+    Use case ends.
+
+**Extensions**:
+
+* 1a. PoochPlanner detects a missing field in the entered input.
+
+   * 1a1. PoochPlanner displays the error message.
+   * 1a2. User re-enters the correct command with a specified field.
+   * Steps 1a1 - 1a2 are repeated until a valid field is inputted by the User.
+   * Use case resumes from step 2. 
+
+* 1b. PoochPlanner detects multiple fields in the entered input.
+
+   * 1b1. PoochPlanner displays the error message.
+   * 1b2. User re-enters a new command with only one field.
+   * Steps 1b1 - 1b2 are repeated until the input references a Person that exists in PoochPlanner.
+   * Use case resumes from step 2. 
+
+* 1c. PoochPlanner detects invalid field in the entered input.
+
+   * 1c1. PoochPlanner displays the error message.
+   * 1c2. User re-enters a new command with another field.
+   * Steps 1c1 - 1c2 are repeated until a valid field is inputted by the User.
+   * Use case resumes from step 2. 
+---
+**System**: `PoochPlanner`
+
+**Use case**: `UC05 - Help Tutorial`
+
+**Actor**: `User`
+
+**MSS**:
+
+1.  User wants to learn more about the commands.
+2.  PoochPlanner displays the information on how to use the different commands.
+
+    Use case ends.
+
+**Extensions**:
+
+* 1a. User wants to learn about available command.
+
+   * 1a1. PoochPlanner displays the list of all commands.
+   * 1a2. User specifies which command they wish to use.
+   * Use case resumes from step 1. 
+
+* 1b. User wants to learn about pooch-staff.
```

Comment:
i think for all cases instead of saying wants to learn you can change to requests to learn

Reply:
Agreed.